packages/core/useDeviceMotion/index.ts

```diff
@@ -31,25 +36,48 @@ export function useDeviceMotion(options: DeviceMotionOptions = {}) {
     z: null,
   })
 
-  if (window) {
-    const onDeviceMotion = createFilterWrapper(
-      eventFilter,
-      (event: DeviceMotionEvent) => {
-        acceleration.value = event.acceleration
-        accelerationIncludingGravity.value = event.accelerationIncludingGravity
-        rotationRate.value = event.rotationRate
-        interval.value = event.interval
-      },
-    )
-
-    useEventListener(window, 'devicemotion', onDeviceMotion)
+  function init() {
+    if (window) {
+      const onDeviceMotion = createFilterWrapper(
+        eventFilter,
+        (event: DeviceMotionEvent) => {
+          acceleration.value = { x: event.acceleration?.x || null, y: event.acceleration?.y || null, z: event.acceleration?.z || null }
+          accelerationIncludingGravity.value = { x: event.accelerationIncludingGravity?.x || null, y: event.accelerationIncludingGravity?.y || null, z: event.accelerationIncludingGravity?.z || null }
+          rotationRate.value = { alpha: event.rotationRate?.alpha || null, beta: event.rotationRate?.beta || null, gamma: event.rotationRate?.gamma || null }
+          interval.value = event.interval
+        },
+      )
+
+      useEventListener(window, 'devicemotion', onDeviceMotion)
+    }
   }
+  const trigger = async () => {
+    if (requirePermission) {
+      const requestPermission = (DeviceMotionEvent as unknown as DeviceMotionEventiOS).requestPermission
+      try {
+        const response = await requestPermission()
+        if (response === 'granted') {
+          permissionGranted.value = true
+          init()
+        }
+      }
+      catch (error) {
+        console.error(error)
+      }
+    }
+  }
+
+  if (!requirePermission)
+    init()
 
   return {
     acceleration,
     accelerationIncludingGravity,
     rotationRate,
     interval,
+    requirePermission,
+    permissionGranted,
```

Comment:
Can we use this convention to keep it consistent? 
https://github.com/vueuse/vueuse/blob/ea87ac59342d0160b908e25bc8ee49bc6c1c2c29/packages/core/useWebNotification/index.ts#L199-L200

Reply:
@antfu done! Please review